lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm_2019/ashrae_90_1_prm_2019.SpaceType.rb

```diff
@@ -0,0 +1,396 @@
+        else
+          user_space_type = user_spacetypes[user_space_type_index]
+          # If multiple LPD value exist - then enforce space-space_type one on one relationship
+          if has_multi_lpd_values_user_data(user_space_type, space_type)
+            space_to_space_type_apply_lighting(user_spaces, user_spacetypes, space_type)
+          else
+            # Process the user_space type data - at this point, we are sure there is no lighting per length
+            # So all the LPD should be identical by space
+            # Loop because we need to assign the occupancy control credit to each space for
+            # Schedule processing.
+            space_type_lighting_per_area = 0.0
+            space_type.spaces.each do |space|
+              space_lighting_per_area = calculate_lpd_from_userdata(user_space_type, space)
+              space_type_lighting_per_area = space_lighting_per_area
+            end
+            space_type.lights[0].lightsDefinition.setWattsperSpaceFloorArea(OpenStudio.convert(space_type_lighting_per_area.to_f, 'W/ft^2', 'W/m^2').get)
+          end
+        end
+      else
+        # no user data, set space_type LPD
+        set_lpd_on_space_type(space_type, user_spaces, user_spacetypes)
+      end
+    end
+  end
+
+  # Function to tset LPD on default space type
+  def set_lpd_on_space_type(space_type, user_spaces, user_spacetypes)
+    if has_multi_lpd_values_space_type(space_type)
+      # If multiple LPD value exist - then enforce space-space_type one on one relationship
+      space_to_space_type_apply_lighting(user_spaces, user_spacetypes, space_type)
+    else
+      # use default - loop through space to assign occupancy credit to each space.
+      space_type_lighting_per_area = 0.0
+      space_type.spaces.each do |space|
+        space_lighting_per_area = calculate_lpd_by_space(space_type, space)
+        space_type_lighting_per_area = space_lighting_per_area
+      end
+      space_type.lights[0].lightsDefinition.setWattsperSpaceFloorArea(OpenStudio.convert(space_type_lighting_per_area.to_f, 'W/ft^2', 'W/m^2').get)
+    end
+  end
+
+  # Function that applies user LPD to each space by duplicating space types
+  # This function is used when there are user space data available or
+  # the spaces under space type has lighting per length value which may cause multiple
+  # lighting power densities under one space_type.
+  # @param user_spaces hash data contained in the user space
+  # @param user_spacetypes hash data contained in the user spacetypes
+  # @param space_type OpenStudio::Model::SpaceType object
+  def space_to_space_type_apply_lighting(user_spaces, user_spacetypes, space_type)
+    space_lighting_per_area_hash = {}
+    # first priority - user_space data
+    if user_spaces && user_spaces.length >= 1
+      space_type.spaces.each do |space|
+        user_space_index = user_spaces.index { |user_space| user_space['name'] == space.name.get }
+        unless user_space_index.nil?
```

Comment:
It's possible that some spaces would have user data at the space level, but not lighting space type userdata (i.e. only WWR bulding type). So instead of "unless user_space_index.nil?", I think this should be something like:

unless user_space_index.nil? num_std_ltg_types = user_data['num_std_ltg_types'].to_i
if num_std_ltg_types > 0

Reply:
added a check